File: Fibonacci.java
package fibonacci;

import java.math.BigInteger;
import java.util.*;


public class Fibonacci {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) 
    {
       
        
        int fib = 0;
        BigInteger a = new BigInteger("0");
        BigInteger b = new BigInteger("1");
        BigInteger c = new BigInteger("0");
        
        while(c.compareTo(BigInteger.ZERO) >= 0 )
        {
            
            System.out.println("fib " + fib);
            
            c = a.add(b);
            a = b;
            b = c;
                
            
            //System.out.println("c " + c);
            fib++;
        }
        
        System.out.println("Fibonacci: " + c);
    }
}

File: Fibonacci.c
#include <stdio.h>

int main()
{
    long fib = 0;

    long long a = 0;
    long long b = 1;
    long long c = 0;

    while(c>=0)
    {
        a=0;
        b=1;
        c=0;
        printf("fib %d \n", fib);
        for(int i=0; i<fib-1; i++)
        {
            c = a+b;
            a = b;
            b = c;
            printf("c %lli \n",c);
        }
        fib++;
    }
    return 0;
}
//short overflows Fibonacci en 24
//int y long overflows Fibonacci en 47
//long long overflows Fibonacci en 93
